src/Api/Controllers/AccountsController.cs

```diff
@@ -285,40 +285,31 @@ public async Task PostKey([FromBody]UpdateKeyRequestModel model)
                 throw new UnauthorizedAccessException();
             }
 
-            var existingCiphers = await _cipherRepository.GetManyByUserIdAsync(user.Id);
-            var ciphersDict = model.Ciphers?.ToDictionary(c => c.Id.Value);
             var ciphers = new List<Cipher>();
-            if (existingCiphers.Any() && ciphersDict != null)
+            if (model.Ciphers.Any())
             {
-                foreach (var cipher in existingCiphers.Where(c => ciphersDict.ContainsKey(c.Id)))
-                {
-                    ciphers.Add(ciphersDict[cipher.Id].ToCipher(cipher));
-                }
+                var existingCiphers = await _cipherRepository.GetManyByUserIdAsync(user.Id);
+                ciphers = existingCiphers
```

Comment:
Awesome, now don't get too annoyed, but since `ciphers` is initialized to `new List<Cipher>()` above, you don't want to just do that for nothing, so either it should be left `null` or unassigned, OR, use `ciphers.AddRange(await _cipherRe....)` here instead of ` ciphers = existingCiphers....` if that makes sense. Not a necessary optimization, but a good one to make.

Reply:
I thought that initializing the empty list was a convenient way to provide a default in case `model.Ciphers.Any()` is false and that `if` block is skipped. In that case, we still need an empty list to pass to `_userService.UpdateKeyAsync` and then to `CipherRepository`. However it is a bit wasteful.

Using `AddRange` seems like it's adding more operations (it would still initialize a new list, the only difference is that it would then join instead of replace the empty list). So instead I assigned `null` as the default instead, and added null checks throughout the subsequent methods. Let me know if this isn't a good solution!

Reply:
@eliykat , I get what you're saying... I think it's better to revert all the downstream null checks and just use `.AddRange()`, all that does is iterate over the passed in `IEnumerable<T>` and adds each to the list. With that, if you remove the `.ToList()` on the line above it (which also iterates the entire `IEnumerable<T>`) then you'll effectively only iterate the list once w/o having to re-assign the variable (win-win).

Reply:
Good point, and much better solution. Thanks!